src/tenantDataMigrations/update-tenant-column-script.js

```diff
@@ -0,0 +1,597 @@
+const { Sequelize } = require('sequelize')
+require('dotenv').config({ path: '.env' })
+
+/**
+ * Script for Tenant Code Migration
+ * - Sets columns to NOT NULL
+ * - Adds primary key constraints with tenant_code
+ * - Adds foreign key relationships
+ * - Configures Citus distribution
+ */
+
+class TenantMigrationFinalizer {
+	constructor() {
+		this.sequelize = new Sequelize(process.env.DEV_DATABASE_URL, {
+			dialect: 'postgres',
+			logging: false,
+			pool: {
+				max: 10,
+				min: 2,
+				acquire: 30000,
+				idle: 10000,
+			},
+		})
+
+		// Tables configuration from helper.js
+		this.allTables = [
+			'availabilities',
+			'connection_requests',
+			'connections',
+			'default_rules',
+			'entities',
+			'entity_types',
+			'feedbacks',
+			'file_uploads',
+			'forms',
+			'issues',
+			'notification_templates',
+			'organization_extension',
+			'post_session_details',
+			'question_sets',
+			'questions',
+			'report_queries',
+			'reports',
+			'resources',
+			'role_extensions',
+			'session_attendees',
+			'session_request',
+			'sessions',
+			'user_extensions',
+		]
+
+		this.tablesWithOrgCode = [
+			'availabilities',
+			'default_rules',
+			'entity_types',
+			'file_uploads',
+			'forms',
+			'issues',
+			'notification_templates',
+			'organization_extension',
+			'question_sets',
+			'questions',
+			'report_queries',
+			'reports',
+			'role_extensions',
+			'user_extensions',
+		]
+
+		this.stats = {
+			constraintsAdded: 0,
+			primaryKeysUpdated: 0,
+			foreignKeysAdded: 0,
+			tablesDistributed: 0,
+			startTime: Date.now(),
+		}
+	}
+
+	/**
+	 * Check if Citus is enabled
+	 */
+	async isCitusEnabled() {
+		try {
+			const [result] = await this.sequelize.query(`
+				SELECT EXISTS(SELECT 1 FROM pg_extension WHERE extname = 'citus') as enabled
+			`)
+			return result[0].enabled
+		} catch (error) {
+			return false
+		}
+	}
+
+	/**
+	 * Check if table is distributed
+	 */
+	async isTableDistributed(tableName) {
+		try {
+			const [result] = await this.sequelize.query(`
+				SELECT COUNT(*) as count 
+				FROM pg_dist_partition 
+				WHERE logicalrelid = '${tableName}'::regclass
+			`)
+			return parseInt(result[0].count) > 0
+		} catch (error) {
+			return false
+		}
+	}
+
+	/**
+	 * Phase 1: Set columns to NOT NULL
+	 */
+	async setColumnsNotNull() {
```

Comment:
@sumanvpacewisdom we need to set after migrations, not begining, how it will run and update the tenant data, did you check this